### PR TITLE
fix(pm): per-check slot key for concurrent master-CI failures

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
## Problem

`derive_slot_key()` collapsed **every** master-CI failure for a repo into one slot key `{repo}#master-ci`. So if a repo's master had **two** failing checks at once (e.g. gptme-cloud had `Conformance Check` AND `test-e2e` red), they shared ONE slot — only one got tracked/dispatched, the other was **invisible** and could fester for hours, blocking every open PR that inherited the red.

**Evidence (2026-06-17):** gptme-cloud master was red on `test-e2e` from PR #421 onward, but the dispatch ledger had zero `master_ci_failure` entries for gptme-cloud that day — only `Conformance Check` entries from 06-16. The test-e2e master-red was never dispatched; it only surfaced when Erik noticed manually.

## Fix

- `derive_slot_key(repo, number, types, check=None)` — add an optional check/workflow name, slugified into the key as `{repo}#master-ci:{check-slug}`. Backward compatible: no check name → bare `{repo}#master-ci`.
- Both `LaneDispatcher` dispatch paths now pass `item.title` (the workflow name).
- Mirrored in the bash twin (`scripts/github/project-monitoring-lib.sh` in ErikBjare/bob): `build_grouped_jsonl` jq grouping now buckets master_ci_failure per check title, and the `derive_slot_key` wrapper passes the check through. Verified end-to-end that bash and Python produce identical slot keys.

## Test

`test_concurrent_master_ci_failures_distinct_slots` + dispatch-level test: two simultaneous master-CI failures on one repo → two distinct slot keys, neither masks the other. Plus slugify edge cases (caps/spaces/punctuation, blank fallback). 277 runloops tests pass.

## Acceptance

Two simultaneous master-CI failures on the same repo produce two distinct slots/dispatches; a long-running failure on check A never prevents detection/dispatch of a new failure on check B.

Tracked: ErikBjare/bob `tasks/pm-master-ci-slot-collapse-fix.md`. Sibling (escalation half): `tasks/master-red-escalation-watchdog.md`.

### CODE REVIEW

File: packages/gptme-runloops/src/gptme_runloops/pm_dispatch.py [hunk 1]
```diff
@@ -247,14 +248,34 @@ def append_full_ledger_entry(
...
+
+    The per-check slug is what keeps two *concurrent* master-CI failures on the
+    same repo (e.g. ``test-e2e`` and ``Conformance Check``) from collapsing into
+    one slot — a long-running failure on check A must never mask detection or
+    dispatch of a new failure on check B.
     """
     if "master_ci_failure" in types:
+        if check and check.strip():
+            return f"{repo}#master-ci:{_slugify_check(check)}"
         return f"{repo}#master-ci"
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply packages/gptme-runloops/src/gptme_runloops/pm_dispatch.py:288]: Fixed in 37c9e4c: non-blank all-punctuation names (e.g. `"---"`) now get a short hash-based slug (`x<6hex>`) instead of falling back to `"unknown"`, preventing the collision. Blank/whitespace names still return `"unknown"` as before. Added `test_master_ci_all_punctuation_check_no_unknown_collision` to cover this case.

**Comment by TimeToBuildBob** [comment]
@greptileai review
